example/plugins/c-api/txn_data_sink/txn_data_sink.c

```diff
@@ -86,20 +90,26 @@ client_reader(TSCont contp, TSEvent event, void *edata)
       TSIOBufferReader reader = TSVIOReaderGet(input_vio);
       int64_t n               = TSIOBufferReaderAvail(reader);
       if (n > 0) {
+        if (data->num_consumed_bytes + n > MAX_BODY_SIZE) {
+          TSError(PCP "Received a response body larger than %d bytes\n", MAX_BODY_SIZE);
```

Comment:
How many times can this error be generated? Should the `VIO` be closed in this case?

Reply:
This actually goes away since this is solved more easily with std::string in c++.